libmamba/src/core/output.cpp

```diff
@@ -399,35 +416,31 @@ namespace mamba
 
     ProgressProxy Console::add_progress_bar(const std::string& name, size_t expected_total)
     {
-        return p_progress_manager->add_progress_bar(name, expected_total);
-    }
-
-    void Console::init_multi_progress(ProgressBarMode mode)
-    {
-        p_progress_manager = make_progress_bar_manager(mode);
+        if (Context::instance().no_progress_bars)
```

Comment:
can we turn this into an assert and make sure that `add_progress_bar` is never called when this is true?

Reply:
Just discussed together it can still make sense for caller code to create a `ProgressProxy` and associate a `ProgressBar` later or just rely on its default constructor (see https://github.com/mamba-org/mamba/blob/d4acbf7dc3d4743c805e8b9085033703f823b9d6/libmamba/include/mamba/core/fetch.hpp#L105).

It makes sense to add `operator bool` to allow syntax like `if(proxy)`, which I added since